cmd/serve/serve.go

```diff
@@ -43,49 +40,68 @@ func NewServeCmd() *cobra.Command {
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			logger.L.Debug("Initializing shutdown channel")
-			c, _ := signal.NotifyContext(cmd.Context(), syscall.SIGINT, syscall.SIGTERM)
+			ctx := cmd.Context()
 
+			// -------------------------------------------------------------------------
+			// GOMAXPROCS
+			_, err := maxprocs.Set(maxprocs.Logger(func(f string, a ...interface{}) { logger.L.Info(fmt.Sprintf(f, a)) }))
+			if err != nil {
+				logger.L.ErrorContext(ctx, "failed to set GOMAXPROCS from cgroups")
+			}
+
+			// -------------------------------------------------------------------------
+			// Print config to stdout
+			logger.L.Info("startup", "server config", serverOpts, "health config", healthOpts)
+
+			shutdown := make(chan os.Signal, 1)
+			signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Is there a benefit of creating a new channel instead using the existing context for signaling?
Wouldn't using the existing context be safer since it would signal everything else running within that context?
``` go
c, _ := signal.NotifyContext(cmd.Context(), syscall.SIGINT, syscall.SIGTERM)
```

Reply:
I want to receive the sys call and `signal.Notify` writes the sys call code to the channel I can get it further down.

I don't know of any better way to do this, I am open to suggestions.

Reply:
The previous code was supposed to do that using the existing context, although it was lacking a `defer stop()`. Not sure if it was working. https://pkg.go.dev/os/signal#NotifyContext 

This seems to be safer as it will signal all the contexts created from the `cmd` one, and not just own our channel.

But I'm fine with discussing this in a different moment.